src/main/kotlin/io/hemin/wien/parser/DateParser.kt

```diff
@@ -1,83 +1,107 @@
 package io.hemin.wien.parser
 
-import java.text.ParsePosition
-import java.text.SimpleDateFormat
-import java.util.Date
+import java.time.LocalDateTime
+import java.time.ZonedDateTime
+import java.time.format.DateTimeFormatter
+import java.time.format.DateTimeFormatterBuilder
+import java.time.format.DateTimeParseException
+import java.time.format.SignStyle
+import java.time.temporal.ChronoField
+import java.time.temporal.TemporalAccessor
 import java.util.Locale
 
 /**
  * Parser implementation for transforming date strings to date objects.
  * Various formats are supported. This class attempts to find the correct
  * format to produce the intended date object.
  */
-internal class DateParser {
+internal object DateParser {
 
-    companion object {
+    private val dayOfWeek = mapOf(
```

Comment:
These are copied from `DateTimeFormatter#RFC_1123_DATE_TIME` — the point is that we don't want to depend on the system locale to be able to parse RFC-(2)822/RFC-1123 timestamps, which are always in English.